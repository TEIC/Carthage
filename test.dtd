<!-- Test.dtd:  a file for testing carthago.
Revisions:
 1997-10-15 : CMSMcQ : made file
-->

<!-- N.B. all comments assume that carthago will be run with the option to
delete all elements 'a e i o u' -->

<!ELEMENT x 
(
  (b | c | d)*,
  ( (a | e)+
    |
    ( 
      (f | g | h)+, 
      (a | e)*
    )
  ),
  (b | c | d)*
)
>

<!-- All of the following should be errors -->
<!ELEMENT x (a) >
<!element x (a?) >
<!element x (a)? >
<!element x (a+) >
<!element x (a)+ >
<!element x (a*) >
<!element x (a)* >
<!element x ((a)) >
<!element x ((a?)) >
<!element x ((a)?) >
<!element x ((a+)) >
<!element x ((a)+) >
<!element x ((a*)) >
<!element x ((a)*) >
<!element x ((((a)))) >
<!element x ((((a?)))) >
<!element x ((((a)?))) >
<!element x ((((a+)))) >
<!element x ((((a)+))) >
<!element x ((((a*)))) >
<!element x ((((a)*))) >
<!-- all of the following should be legal -->
<!element x (b) >
<!element x (b?) >
<!element x (b)? >
<!element x (b+) >
<!element x (b)+ >
<!element x (b*) >
<!element x (b)* >
<!element x ((b)) >
<!element x ((b?)) >
<!element x ((b)?) >
<!element x ((b+)) >
<!element x ((b)+) >
<!element x ((b*)) >
<!element x ((b)*) >
<!element x ((((b)))) >
<!element x ((((b?)))) >
<!element x ((((b)?))) >
<!element x ((((b+)))) >
<!element x ((((b)+))) >
<!element x ((((b*)))) >
<!element x ((((b)*))) >
<!-- all of the following should leave one required element -->
<!element x (a) >
<!element x (a?, a) >
<!element x (a?, a, (a, a?)?) >
<!element x (a | a) >
<!element x (o? | e+ | (i+)? | a) >
<!element x (o? | e+ | (i+)? | a | b) >
<!element x (e? & i* & a & (u | o+)*) >
<!element x (b) >
<!element x (a?, b) >
<!element x (a?, b, (e, e?)?) >
<!element x (b | u) >
<!element x (o? | e+ | (i+)? | b) >
<!element x (o? | e+ | (i+)? | a | b) >
<!element x (e? & i* & b & (u | o+)*) >
<!-- all of the following should result in an optional element -->
<!element x (a?) >
<!element x ((((a))?)) >
<!element x ((((a)?)?))? >
<!element x (e | a?) >
<!element x ((((b | e | a))?)) >
<!element x ((((a*, e)?)?))? >
<!element x (b?) >
<!element x ((((b))?)) >
<!element x ((((b)?)?))? >
<!element x (e | b?) >
<!element x ((((b | e | a))?)) >
<!element x ((((a*, b)?)?))? >
<!element x ((((a*, b)?, (a,e,i,o+,u+)*, e?)?))? >

<!-- all of the following should result in a single element with a star -->
<!element x (a*) >
<!element x ((((a))?))+ >
<!element x ((((a)?)+))? >
<!element x (e | a?)* >
<!element x ((((b+ | e+ | a+))?)) >
<!element x ((((a*, e*)?)?))? >
<!element x (b+)* >
<!element x ((((b?)+)?)) >
<!element x ((((b)?)?)+)? >
<!element x (e | b?)+ >
<!element x ((((b+ | e | a))?)) >
<!element x ((((a*, b)?)+))? >
<!element x ((((a*, b)?, (a,e,i,o+,u+)*, e?)?))* >
