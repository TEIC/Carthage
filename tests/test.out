<!-- Test.dtd:  a file for testing carthago.
Revisions:
 1997-10-15 : CMSMcQ : made file
-->
<!-- N.B. all comments assume that carthago will be run with the option to
delete all elements 'a e i o u' -->

<!ELEMENT x  ((b | c | d)*, (f | g | h)+, (b | c | d)*) >

<!-- All of the following should be errors -->

<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (ZOMBIE...a)? >


<!ELEMENT x  (ZOMBIE...a)? >


<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (ZOMBIE...a)+ >


<!ELEMENT x  (ZOMBIE...a)* >


<!ELEMENT x  (ZOMBIE...a)* >


<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (ZOMBIE...a)? >


<!ELEMENT x  (ZOMBIE...a)? >


<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (ZOMBIE...a)+ >


<!ELEMENT x  (ZOMBIE...a)* >


<!ELEMENT x  (ZOMBIE...a)* >


<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (ZOMBIE...a)? >


<!ELEMENT x  (ZOMBIE...a)? >


<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (ZOMBIE...a)+ >


<!ELEMENT x  (ZOMBIE...a)* >


<!ELEMENT x  (ZOMBIE...a)* >

<!-- all of the following should be legal -->

<!ELEMENT x  (b) >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)+ >


<!ELEMENT x  (b)+ >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b) >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)+ >


<!ELEMENT x  (b)+ >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b) >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)+ >


<!ELEMENT x  (b)+ >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b)* >

<!-- all of the following should leave one required element -->

<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (b) >


<!ELEMENT x  (ZOMBIE...a) >


<!ELEMENT x  (b) >


<!ELEMENT x  (b) >


<!ELEMENT x  (b) >


<!ELEMENT x  (b) >


<!ELEMENT x  (b) >


<!ELEMENT x  (b) >


<!ELEMENT x  (b) >

<!-- all of the following should result in an optional element -->

<!ELEMENT x  (ZOMBIE...a)? >


<!ELEMENT x  (ZOMBIE...a)? >


<!ELEMENT x  (ZOMBIE...a)? >


<!ELEMENT x  (ZOMBIE...a)? >


<!ELEMENT x  (b)? >


<!ELEMENT x  (ZOMBIE...e)? >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)? >


<!ELEMENT x  (b)? >

<!-- all of the following should result in a single element with a star -->

<!ELEMENT x  (ZOMBIE...a)* >


<!ELEMENT x  (ZOMBIE...a)* >


<!ELEMENT x  (ZOMBIE...a)* >


<!ELEMENT x  (ZOMBIE...a)* >


<!ELEMENT x  (b)* >


<!ELEMENT x  (ZOMBIE...e)* >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b)* >


<!ELEMENT x  (b)* >

